presto-hive/src/main/java/com/facebook/presto/hive/rule/HivePartialAggregationPushdown.java

```diff
@@ -255,6 +257,10 @@ private Optional<PlanNode> tryPartialAggregationPushdown(PlanNode plan)
                     RowExpression column = callExpression.getArguments().get(0);
                     columnName = column.toString();
                     HiveColumnHandle oldColumnHandle = (HiveColumnHandle) oldTableScanNode.getAssignments().get(column);
+                    // These columns get 'PREFILLED' with values in the corresponding page sources
+                    if (oldColumnHandle.getColumnType() == PARTITION_KEY || oldColumnHandle.getColumnType() == SYNTHESIZED) {
```

Comment:
I would check instead that the column is regular type. That way it won't break if there's some other kind of column that we add later that shows up.  

Also, would it be possible to move this check to isAggregationPushdownSupported?  It would make the flow a bit easier to follow to have all the checks in one place, and easier to see the cases where the optimizer will be invoked.